pkg/network/events/monitor.go

```diff
@@ -54,13 +78,44 @@ func UnregisterHandler(handler ProcessEventHandler) {
 
 type eventHandlerWrapper struct{}
 
-func (h *eventHandlerWrapper) HandleEvent(ev *model.Event) {
+func (h *eventHandlerWrapper) HandleEvent(ev interface{}) {
+	if ev == nil {
+		log.Errorf("Received nil event")
+		return
+	}
+
+	evProcess, ok := ev.(*Process)
+	if !ok {
+		log.Errorf("Event is not a process")
+		return
+	}
+
 	m := theMonitor.Load()
 	if m != nil {
-		m.(*eventMonitor).HandleEvent(ev)
+		m.(*eventMonitor).HandleEvent(evProcess)
 	}
 }
 
+// Copy copies the necessary fields from the event received from the event monitor
+func (h *eventHandlerWrapper) Copy(ev *model.Event) interface{} {
```

Comment:
Should we use generics to avoid returning `any`?

Reply:
i'm gonna try that in a follow-up PR! thanks for the suggestion!

Reply:
Discussed offline - this is okay to do in a follow-up